src/ConcurrentUtilities.jl

```diff
@@ -4,6 +4,14 @@ export Lockable, OrderedSynchronizer, reset!, ReadWriteLock, readlock, readunloc
     Workers, remote_eval, remote_fetch, Worker, terminate!, WorkerTerminatedException,
     Pool, acquire, release, drain!, try_with_timeout, TimeoutException
 
+macro samethreadpool_spawn(expr)
+    if isdefined(Base.Threads, :threadpool)
+        esc(:(Threads.@spawn Threads.threadpool() $expr))
```

Comment:
@kpanmay, do you know why `Threads.@spawn` doesn't do this by default?

Reply:
The thinking was that the default behavior should be to spawn to the default thread pool and you need to explicitly spawn an interactive task because they have an implicit contract to not hog the CPU. For backward compatibility, the contract cannot be assumed.